src/Compilers/Core/CodeAnalysisTest/Collections/TopologicalSortTests.cs

```diff
@@ -0,0 +1,204 @@
+            AssertTopologicallySorted(sorted, successors, "Test01");
+            Assert.Equal(6, sorted.Length);
+            Assert.Equal(4, sorted[0]);
+            Assert.Equal(5, sorted[1]);
+            Assert.Equal(2, sorted[2]);
+            Assert.Equal(3, sorted[3]);
+            Assert.Equal(1, sorted[4]);
+            Assert.Equal(0, sorted[5]);
+        }
+
+        [Fact]
+        public void Test02()
+        {
+            int[][] successors = new int[][]
+            {
+                /* 0 */ new int[] { },
+                /* 1 */ new int[] { 2, 4 },
+                /* 2 */ new int[] { },
+                /* 3 */ new int[] { 2, 5 },
+                /* 4 */ new int[] { 2, 3 },
+                /* 5 */ new int[] { 2, },
+                /* 6 */ new int[] { 2, 7 },
+                /* 7 */ new int[] { }
+            };
+
+            var sorted = TopologicalSort.IterativeSort<int>(new[] { 1, 6 }, x => successors[x]);
+            AssertTopologicallySorted(sorted, successors, "Test02");
+            Assert.Equal(7, sorted.Length);
+            Assert.Equal(1, sorted[0]);
+            Assert.Equal(4, sorted[1]);
+            Assert.Equal(3, sorted[2]);
+            Assert.Equal(5, sorted[3]);
+            Assert.Equal(6, sorted[4]);
+            Assert.Equal(7, sorted[5]);
+            Assert.Equal(2, sorted[6]);
+        }
+
+        [Fact]
+        public void TestCycle()
+        {
+            int[][] successors = new int[][]
+            {
+                /* 0 */ new int[] { },
+                /* 1 */ new int[] { 2, 4 },
+                /* 2 */ new int[] { },
+                /* 3 */ new int[] { 2, 5 },
+                /* 4 */ new int[] { 2, 3 },
+                /* 5 */ new int[] { 2, 1 },
+                /* 6 */ new int[] { 2, 7 },
+                /* 7 */ new int[] { }
+            };
+
+            // 1 -> 4 -> 3 -> 5 -> 1
+            Assert.Throws<ArgumentException>(() =>
+            {
+                var sorted = TopologicalSort.IterativeSort<int>(new[] { 1 }, x => successors[x]);
+            });
+        }
+
+        [Theory]
+        [InlineData(1984142830)]
+        [InlineData(107329897)]
+        [InlineData(136826316)]
+        [InlineData(808774716)]
+        [InlineData(729791148)]
+        [InlineData(770911997)]
+        [InlineData(1786285961)]
+        [InlineData(321110113)]
+        [InlineData(1686926633)]
+        [InlineData(787934201)]
+        [InlineData(745939035)]
+        [InlineData(1075862430)]
+        [InlineData(428872484)]
+        [InlineData(489337268)]
+        [InlineData(1976108951)]
+        [InlineData(428397397)]
+        [InlineData(1921108202)]
+        [InlineData(926330127)]
+        [InlineData(364136202)]
+        [InlineData(1893034696)]
+        public void TestRandom(int seed)
+        {
+            int numberOfNodes = 100;
+            Random random = new Random(seed);
```

Comment:
Are we sure Random works the same between platforms/versions etc?

Reply:
Just checked the docs. It seems to be specified as `Donald E. Knuth's subtractive random number generator`